catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -1142,68 +1149,79 @@ private ElementSetNameType createElementSetName(ElementSetType type) {
-                    .getPropertyName() != null) {
-                propName = query.getSortBy()
-                        .getPropertyName()
-                        .getPropertyName();
+    private SortByType createSortBy(QueryRequest queryRequest) {
 
-                if (propName != null) {
-                    if (Result.TEMPORAL.equals(propName) || Metacard.ANY_DATE.equals(propName)) {
-                        propName = Core.MODIFIED;
-                    } else if (Result.RELEVANCE.equals(propName) || Metacard.ANY_TEXT.equals(
-                            propName)) {
-                        propName = Core.TITLE;
-                    } else if (Result.DISTANCE.equals(propName)
-                            || Metacard.ANY_GEO.equals(propName)) {
-                        return null;
-                    }
+        Query query = queryRequest.getQuery();
+        SortByType cswSortBy = null;
+
+        if (query != null && query.getSortBy() != null && query.getSortBy()
+                .getPropertyName() != null) {
+            List<SortBy> sortBys = new ArrayList<>();
+            sortBys.add(query.getSortBy());
+            Serializable extSortBySer = queryRequest.getPropertyValue(EXT_SORT_BY);
+            if (extSortBySer instanceof SortBy[]) {
+                SortBy[] extSortBys = (SortBy[]) extSortBySer;
+                if (extSortBys.length > 0) {
+                    sortBys.addAll(Arrays.asList(extSortBys));
+                }
+            }
 
-                    propertyName.setContent(Arrays.asList((Object) cswFilterDelegate.mapPropertyName(
-                            propName)));
-                    sortProperty.setPropertyName(propertyName);
-                    if (SortOrder.DESCENDING.equals(query.getSortBy()
-                            .getSortOrder())) {
-                        sortProperty.setSortOrder(SortOrderType.DESC);
-                    } else {
-                        sortProperty.setSortOrder(SortOrderType.ASC);
+            cswSortBy = new SortByType();
+            for (SortBy sortBy : sortBys) {
+                SortPropertyType sortProperty = new SortPropertyType();
+                PropertyNameType propertyName = new PropertyNameType();
+
+                if (sortBy.getPropertyName() != null) {
+                    String propName = sortBy.getPropertyName()
+                            .getPropertyName();
+
+                    if (propName != null) {
+                        if (Result.TEMPORAL.equals(propName) || Metacard.ANY_DATE.equals(propName)) {
+                            propName = Core.MODIFIED;
+                        } else if (Result.RELEVANCE.equals(propName) || Metacard.ANY_TEXT.equals(
+                                propName)) {
+                            propName = Core.TITLE;
+                        } else if (Result.DISTANCE.equals(propName) || Metacard.ANY_GEO.equals(
+                                propName)) {
+                            return null;
```

Comment:
❓ If there are multiple sortBys and one of them is distance, do you really want to return null for all of the other sortBys?  Can we just skip distance and anyGeo instead?